.github/workflows/ci.yml

```diff
@@ -89,21 +89,49 @@ jobs:
       - uses: taiki-e/install-action@v2
         with:
           tool: nextest
-      - name: "Test (nextest)"
+      - name: Test (nextest)
         id: nextest
         env:
-          GIX_TEST_IGNORE_ARCHIVES: 1
+          GIX_TEST_IGNORE_ARCHIVES: '1'
         run: cargo nextest --profile=with-xml run --workspace --no-fail-fast
         continue-on-error: true
-      - name: Check how many tests failed
-        if: steps.nextest.outcome == 'failure'
+      - name: Check for errors
+        run: |
+          [xml]$junit_xml = Get-Content -Path 'target/nextest/with-xml/junit.xml'
+          if ($junit_xml.testsuites.errors -ne 0) { exit 1 }
+      - name: Collect actual failures
+        run: |
+          [xml]$junit_xml = Get-Content -Path 'target/nextest/with-xml/junit.xml'
+
+          $actual_failures = $junit_xml.SelectNodes("//testcase[failure]") |
+              ForEach-Object { "$($_.classname) $($_.name)" } |
+              Sort-Object
+
+          Write-Output $actual_failures
+          Set-Content -Path 'actual-failures.txt' -Value $actual_failures
+      - name: Collect expected failures
         env:
-          # See https://github.com/GitoxideLabs/gitoxide/issues/1358.
-          EXPECTED_FAILURE_COUNT: 14
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          $issue = 1358  # https://github.com/GitoxideLabs/gitoxide/issues/1358
```

Comment:
This approach, as interesting as it may seem, breaks isolation of the test a bit too much for my taste and I'd prefer hardcoding it. The issue could then refer to the file to display the lines here, or link to the separate file that is read by the workflow.

Reply:
Sounds good. I've pushed [99238a7](https://github.com/GitoxideLabs/gitoxide/pull/1663/commits/99238a72894595e95d8c03df51fda02285447589), which adds a file with the list and has the job use that, removing altogether the retrieval step.

A subtlety here is that the list of expected failures is no longer being sorted. I can do that if desired, but it seemed like it would be preserving unnecessary complexity. Even if for some strange reason `cargo nextest` when run in manual testing were to output results not in this order, this is a file being committed to the repository, so the entries can be ordered lexicographically when committing changes to them.

(The list of actual failures is still sorted, so there is no doubt that the CI test run itself is producing output that is ready for comparison.)

I'm not sure what the file should be called, but this gigantic name seemed reasonable to me, since it makes everything clear. What I'm less sure of is it's location, but the existing `etc` directory seemed like it might be the best place.